prov/cxi/src/cxip_cq.c

```diff
@@ -511,7 +511,8 @@ int cxip_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 	}
 
 	cxi_cq->util_cq.cq_fid.fid.ops = &cxip_cq_fi_ops;
-	cxi_cq->util_cq.cq_fid.ops = &cxip_cq_ops;
+	if (!(attr->flags & FI_PEER))
+		cxi_cq->util_cq.cq_fid.ops = &cxip_cq_ops;
```

Comment:
The change looks good. However, could add a check in cxip_cq_verify_attr() to return -FI_ENOSYS if the FI_PEER flag is set and the attr->wait_obj is not FI_WAIT_NONE. Our WAIT_FD wait object is reliant on the cxi specific CQ sread/sreadfrom. This might avoid a future incompatibility/error.

Reply:
@amirshehataornl - I'm good with this patch once Steve's comment is addressed. Thanks!

Reply:
@swelch, let me know if the changes look alright to you.

Reply:
Thank you